packages/experiments/src/components/Tile/Tile.tsx

```diff
@@ -158,6 +158,55 @@ export class Tile extends BaseComponent<ITileProps, ITileState> {
     const isSelectable = !!selection && selectionIndex > -1;
     const isInvokable = (!!href || !!onClick || !!invokeSelection) && !isModal;
 
+    const content = (
+      <>
+        {ariaLabel ? (
+          <span key="label" id={this._labelId} className={css('ms-Tile-label', TileStylesModule.label)}>
+            {ariaLabel}
+          </span>
+        ) : null}
+        {background
+          ? this._onRenderBackground({
+              background: background,
+              hideBackground
+            })
+          : null}
+        {foreground
+          ? this._onRenderForeground({
+              foreground: foreground,
+              hideForeground
+            })
+          : null}
+        {itemName || itemActivity
+          ? this._onRenderNameplate({
+              name: itemName,
+              activity: itemActivity
+            })
+          : null}
+      </>
+    );
+
+    const link = href ? (
+      <a
+        href={href}
+        onClick={onClick}
+        ref={this.props.linkRef}
+        data-selection-invoke={isInvokable && selectionIndex > -1 ? true : undefined}
+        className={css('ms-Tile-link', TileStyles.link)}
+      >
+        {content}
+      </a>
+    ) : (
+      <button
+        onClick={onClick}
+        ref={this.props.linkRef}
+        data-selection-invoke={isInvokable && selectionIndex > -1 ? true : undefined}
+        className={css('ms-Tile-link', TileStyles.link)}
+      >
+        {content}
+      </button>
+    );
```

Comment:
Could this just be:

```suggestion
    const ComponentAs = href ? 'a' : 'button';
    const link = (
      <ComponentAs
        href={href}
        onClick={onClick}
        ref={this.props.linkRef}
        data-selection-invoke={isInvokable && selectionIndex > -1 ? true : undefined}
        className={css('ms-Tile-link', TileStyles.link)}
      >
        {content}
      </a>
    );
```

Reply:
Type-safety won't allow the `href` element when `ComponentAs` is `'button'`, though.

Reply:
Sorry @JasonGore, I got a little trigger happy and should have given you a chance to respond before merging this. Want me to open a new issue for this?

Reply:
Thomas covered it in #7746, thanks 😄

Reply:
Nice teamwork all!